File: benchmarks/fixtures/java/semantic-cases/src/main/java/com/example/app/Runner.java
package com.example.app;

import com.example.services.FriendlyGreeter;
import com.example.services.Greeter;
import static com.example.util.Names.defaultName;

public class Runner extends BaseRunner implements Runnable {
    private final Greeter greeter;

    public Runner(Greeter greeter) {
        this.greeter = greeter;
    }

    @Override
    public void run() {
        String name = prepareName(defaultName());
        greeter.greet(name);
        new Helper().assist();
    }

    public static Runner buildDefault() {
        return new Runner(new FriendlyGreeter());
    }
}

class BaseRunner {
    protected String prepareName(String raw) {
        return raw.trim();
    }
}

record Helper() {
    void assist() {}
}

File: benchmarks/fixtures/kotlin/semantic-cases/src/main/kotlin/com/example/app/Runner.kt
package com.example.app

import com.example.services.FriendlyGreeter as Friendly
import com.example.services.Greeter
import com.example.util.fetchDefault
import com.example.util.prepare

class Runner(private val greeter: Greeter) {
    suspend fun run(): String {
        val name = "world".prepare()
        val default = fetchDefault()
        return greeter.greet(default) + name
    }

    companion object {
        fun buildDefault(): Runner = Runner(Friendly.create())
    }
}
